### PR TITLE
add standard label for the filtering of cluster

### PR DESCRIPTION
**Pull request by scarlet25151**
## Related issue number

#338 

## Checks

- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [ ] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by Jeffwan** [comment]
I create #338 for other reason. I will cut a PR to add labels to manifest.
I think what you need to do here is to make sure apiserver just filter out the cluster with some label like `app.kubernetes.io/managed-by=kuberay-apiserver`

File: ray-operator/controllers/ray/common/constant.go [hunk 1]
```diff
@@ -23,6 +26,12 @@ const (
 	DefaultDashboardName  = "dashboard"
 	DefaultMetricsName    = "metrics"
 
+	// The default application name
+	DefaultApplicationName = "kuberay"
+
+	// The default name for raycluster controller
+	DefaultRayClusterControllerName = "raycluster-controller"
```
Comment (Jeffwan) [inline ray-operator/controllers/ray/common/constant.go:33]: check https://github.com/ray-project/kuberay/pull/345 for naming

File: ray-operator/controllers/ray/common/constant.go [hunk 2]
```diff
@@ -9,6 +9,9 @@ const (
 	RayNodeLabelKey      = "ray.io/is-ray-node"
 	RayIDLabelKey        = "ray.io/identifier"
 
+	DefaultAppNameKey      = "app.kubernetes.io/name"
```
Comment (Jeffwan) [inline ray-operator/controllers/ray/common/constant.go:12]: This won't be configurable and so it's remove `Default` prefix

File: ray-operator/controllers/ray/common/ingress.go [hunk 3]
```diff
@@ -17,8 +17,10 @@ const IngressClassAnnotationKey = "kubernetes.io/ingress.class"
 // This is used to expose dashboard for external traffic.
 func BuildIngressForHeadService(cluster rayiov1alpha1.RayCluster) (*networkingv1.Ingress, error) {
 	labels := map[string]string{
-		RayClusterLabelKey: cluster.Name,
-		RayIDLabelKey:      utils.GenerateIdentifier(cluster.Name, rayiov1alpha1.HeadNode),
+		RayClusterLabelKey:     cluster.Name,
+		RayIDLabelKey:          utils.GenerateIdentifier(cluster.Name, rayiov1alpha1.HeadNode),
+		DefaultAppNameKey:      DefaultApplicationName,
+		DefaultAppComponentKey: DefaultRayClusterControllerName,
```
Comment (Jeffwan) [inline ray-operator/controllers/ray/common/ingress.go:23]: the data plane workloads doesn't need to have `DefaultAppComponentKey`?  you can use `app.kubernetes.io/managed-by` for all resources managed by operators. for the ray cluster, use `app.kubernetes.io/managed-by=kuberay-apiserver`

**Review by Jeffwan — CHANGES_REQUESTED** [review]

File: ray-operator/controllers/ray/common/constant.go [hunk 4]
```diff
@@ -9,6 +9,9 @@ const (
 	RayNodeLabelKey      = "ray.io/is-ray-node"
 	RayIDLabelKey        = "ray.io/identifier"
 
+	NameLabelKey      = "app.kubernetes.io/name"
```
Comment (Jeffwan) [inline ray-operator/controllers/ray/common/constant.go:12]: KubernetesApplicationNameLabelKey
KubernetesCreatedByLabelKey

same naming for `apiserver/pkg/util/config.go`

File: apiserver/pkg/util/cluster.go [hunk 5]
```diff
@@ -12,6 +12,14 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+const (
+	// The default application name
+	DefaultApplicationName = "kuberay"
+
+	// The default name for apiserver
+	DefaultAPIServerName = "kuberay-apiserver"
```
Comment (Jeffwan) [inline apiserver/pkg/util/cluster.go:20]: Change to `ComponentName`

File: apiserver/pkg/util/cluster.go [hunk 6]
```diff
@@ -12,6 +12,14 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+const (
+	// The default application name
+	DefaultApplicationName = "kuberay"
```
[the remark this answers is not in the document]
Comment (scarlet25151) [reply apiserver/pkg/util/cluster.go]: Resolved. move const definition to config.go

**Comment by Jeffwan** [comment]
It looks good to me now

**Review by Jeffwan — APPROVED** [review]

**Pull request merged**